packages/superset-ui-chart-controls/src/types.ts

```diff
@@ -253,28 +272,23 @@ export type SharedSectionAlias =
   | 'sqlaTimeSeries'
   | 'NVD3TimeSeries';
 
-export interface OverrideSharedControlItem {
-  name: SharedControlAlias;
-  override: Partial<SharedControlConfig>;
+export interface OverrideSharedControlItem<A extends SharedControlAlias = SharedControlAlias> {
+  name: A;
+  override: Partial<SharedControls[A]>;
 }
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export interface CustomControlItem<P = any> {
+export type CustomControlItem = {
   name: string;
-  config: CustomComponentControlConfig<P>;
-}
-
-export type ControlSetItem =
-  | SharedControlAlias
-  | OverrideSharedControlItem
-  | CustomControlItem
-  // use ReactElement instead of ReactNode because `string`, `number`, etc. may
-  // interfere with other ControlSetItem types
-  | ReactElement
-  | null;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  config: BaseControlConfig<any, any, any>;
```

Comment:
The typing is so tricky. Had to use a lot of `any`s to make it work. Happy to revise if people have better suggestions.

Reply:
Would `unknown` work or does it have to be `any`?

Reply:
It has to be `any`, as it is used as a catch-all type in `ControlSetRow`. `unknown` in instantiated generic types are not compatible with another instance of the same generic type.